slsDetectorSoftware/include/multiSlsDetector.h

```diff
@@ -622,6 +622,20 @@ class multiSlsDetector : public virtual slsDetectorDefs {
      * @returns OK or FAIL
      */
     int configureMAC(int detPos = -1);
+    
+     /**
+     * Set starting frame number for the next acquisition
+     * @param val starting frame number
+     * @param detPos -1 for all detectors in  list or specific detector position
+     */
+    void setStartingFrameNumber(uint64_t value, int detPos = -1);
+
+     /**
+     * Get starting frame number for the next acquisition
+     * @param detPos -1 for all detectors in  list or specific detector position
+     * @returns starting frame number
+     */
+    uint64_t getStartingFrameNumber(int detPos = -1);
```

Comment:
could be const

Reply:
Made the argument of setStartingFrameNumber const. Could not make getStartingFrameNumber const as it sends the arguments to the detector.

Regarding servers, I completely agree. It is very error prone as it is at the moment, which was also why I preferred C++. We can look into it later.